File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/robotservices/maps/MapsServiceFactory.java
package com.neatorobotics.sdk.android.robotservices.maps;

/**
 * Neato
 * Created by Marco on 04/01/2017.
 * Copyright © 2016 Neato Robotics. All rights reserved.
 */

public class MapsServiceFactory {
    public static MapsService get(String serviceVersion) {
        if("advanced-1".equalsIgnoreCase(serviceVersion)) return new MapsAdvanced1Service();
        else if("macro-1".equalsIgnoreCase(serviceVersion)) return new MapsMacro1Service();
        else if("basic-2".equalsIgnoreCase(serviceVersion)) return new MapsBasic2Service();
        else if("basic-1".equalsIgnoreCase(serviceVersion)) return new MapsBasic1Service();
        else return null;
    }
}

File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/robotservices/maps/MapsServiceFactory.kt
/*
 * Copyright (c) 2019.
 * Neato Robotics Inc.
 */

package com.neatorobotics.sdk.android.robotservices.maps

import com.neatorobotics.sdk.android.robotservices.RobotServices

object MapsServiceFactory {
    fun get(serviceVersion: String): MapsService? {
        return when {
            RobotServices.VERSION_BASIC_1.equals(serviceVersion, ignoreCase = true) -> MapsBasic1Service()
            RobotServices.VERSION_ADVANCED_1.equals(serviceVersion, ignoreCase = true) -> MapsAdvanced1Service()
            RobotServices.VERSION_MACRO_1.equals(serviceVersion, ignoreCase = true) -> MapsMacro1Service()
            RobotServices.VERSION_BASIC_2.equals(serviceVersion, ignoreCase = true) -> MapsBasic2Service()
            else -> null
        }
    }
}
